### ISSUE TITLE
indent: false positives for objects in array variable declaration

### ISSUE THREAD

**Issue by jwilsson** [eslint/eslint#5586]
**What version of ESLint are you using?**
2.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{
  "extends": "eslint:recommended",
  "env": {
    "mocha": true,
    "node": true
  },
  "rules": {
    "array-bracket-spacing": 2,
    "brace-style": 2,
    "camelcase": 2,
    "comma-dangle": 0,
    "comma-spacing": 2,
    "comma-style": 2,
    "computed-property-spacing": 2,
    "curly": 2,
    "dot-location": [2, "property"],
    "dot-notation": 2,
    "eol-last": 2,
    "eqeqeq": 2,
    "guard-for-in": 2,
    "indent": [2, 4, {
        "SwitchCase": 1,
        "VariableDeclarator": 0
    }],
    "key-spacing": 2,
    "keyword-spacing": [2, {
        "after": true,
        "before": true
    }],
    "newline-after-var": 2,
    "no-caller": 2,
    "no-console": 0,
    "no-else-return": 2,
    "no-eval": 2,
    "no-extend-native": 2,
    "no-lonely-if": 2,
    "no-loop-func": 2,
    "no-multi-spaces": 2,
    "no-trailing-spaces": 2,
    "one-var": [2, "never"],
    "quotes": [2, "single", "avoid-escape"],
    "semi": 2,
    "space-before-blocks": 2,
    "space-before-function-paren": 2,
    "space-in-parens": 2,
    "space-unary-ops": [2, {
        "nonwords": false,
        "words": true
    }],
    "strict": [2, "global"],
    "vars-on-top": 2,
    "yoda": 2
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
Before `2.4.0` the following code didn't report any errors:
```js
'use strict';

var lmn = [{
    a: 1
},
{
    b: 2
},
{
    x: 2
}];

console.log(lmn);
```

but as of `2.4.0` is does (output below).

However, this code works fine (even on `2.4.0`):
```js
'use strict';

var lmn;

lmn = [{
    a: 1
},
{
    b: 2
},
{
    x: 2
}];

console.log(lmn);
```

**What did you expect to happen?**
I expected the first code example to go through without any errors.

**What actually happened? Please include the actual, raw output from ESLint.**
```
   6:1  error  Expected indentation of 4 space characters but found 0  indent
   7:5  error  Expected indentation of 8 space characters but found 4  indent
   8:1  error  Expected indentation of 4 space characters but found 0  indent
   9:1  error  Expected indentation of 4 space characters but found 0  indent
  10:5  error  Expected indentation of 8 space characters but found 4  indent
  11:1  error  Expected indentation of 4 space characters but found 0  indent
```

I think I managed to track it down to the change introduced in #5509, removing that change fixes this issue.

**Comment by gyandeeps** [comment]
With the recent changes, we introduced indent check inside the variable declaration to check for proper indent. so your code would become something like this

```js
var lmn = [
    {
        a: 1
    },
    {
        b: 2
    },
    {
        x: 2
    }
];

console.log(lmn);
```

Handling scenarios like this has become overly complicated because there are so many variation of this. I think at this point whenever we fix one scenarios some other scenario which is not in our tests breaks.

**Comment by platinumazure** [comment]
@gyandeeps Do you think this is a bug? Or is this now as designed?

**Comment by TJKresch** [comment]
@gyandeeps That solution looks right to me. Multiline array literals are more similar to block-scopes and object-literals than multiline variable declarations. I was previously forced to remove my VariableDeclarator indentation rules for this exact reason, but that solution you described would fix my problem.

**Comment by ilyavolodin** [comment]
I verified that code posted by @gyandeeps is linting without errors, and it looks correctly indented to me. Closing, as it looks like the rule is behaving correctly in this case. If you disagree, feel free to reopen.
